monolith/src/lib.rs

```diff
@@ -0,0 +1,7 @@
+//! The Monolith permutation, and hash functions built from it.
```

Comment:
Couple minor things,
- can this be `#![no_std]`?
- I think private modules with `pub use` would be nice - it looks like there are only two structs so might as well put them both in the module root

Reply:
Done!